### PR TITLE
add backend and load dummy data

### LINKED ISSUE #31: backend: load dummy data
## Value statement
As a **developer** 
I want to **load dummy data from a MongoDB** 
so that **I can remove the hard-coded testArray from my code** 

## Description
Clicking on a "load dummy data"-Button fetches and renders a guestList and a favoriteRecipeList. 

## Acceptance criteria
- [x] there is a "load dummy data"-button on home page
- [x] clicking on the button loads dummy-data from a MongoDB
- [ ] a dummy guestList and a dummy favoriteRecipeList are rendered then 

## Tasks
- [x] add a "load-dummy data"-button to home
- [x] set up a MongoDB
- [x] add dummy data to MongoDB
- [x] add button-function to load dummy data from MongoDB

## Size
medium

### PR DESCRIPTION
**Pull request by s-kond**
fixes #31 
Took a little longer than expected - but fetching dummy data from my MongoDB should work now!

The UserModel-File and the "GET /:id"- and "POST"-route can't be used from the UI at this point, but that will be the next step.

### CODE REVIEW

File: api/models/UserModel.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        name: String,
+        notes: String,
+        selected: Boolean,
+      },
+    ],
+  },
+  { timestamps: true }
+);
+
+module.exports = mongoose.model("User", userSchema);
```
Comment (Kheirah) [inline api/models/UserModel.js]:
```suggestion
module.exports = mongoose.models.User || mongoose.model("User", userSchema);
```

File: src/App.js [hunk 2]
```diff
@@ -44,6 +24,20 @@ function App() {
...
   const navigate = useNavigate();
 
+  async function fetchGuestList() {
+    try {
+      const response = await fetch("/api/users/all");
+      const data = await response.json();
+
+      if (response.ok) {
+        const guestList = data.map((user) => user.guestList);
+        setGuestArray([...guestArray, ...guestList[0]]);
```
Comment (Kheirah) [inline src/App.js:34]: I'm potentially too dazed to really tell, but I'm not sure why you are spreading the `guestList` on index 0 here. Maybe you can check for yourself whether or not that actually makes sense.

**Review by Kheirah — APPROVED** [review]
Splendid!

File: src/App.js (hunk 2 above)
Comment (s-kond) [reply src/App.js:34]: It was a long day - I'm pretty dazed, too... 
I checked it with console.logs until it worked and that's the result 😅 
Makes sense for now, but I'll probably change it during one of the next steps...

**Pull request merged**